pkg/types/chain_reader.go

```diff
@@ -41,8 +41,25 @@ type ChainReader interface {
 	GetLatestValue(ctx context.Context, contractName string, method string, params, returnVal any) error
 
 	// Bind will override current bindings for the same contract, if one has been set and will return an error if the
-	// contract is not known by the ChainReader, or if the address is invalid
+	// contract is not known by the ChainReader, or if the Address is invalid
 	Bind(ctx context.Context, bindings []BoundContract) error
+
+	//TODO Rebind binding address
+	//ReBind(ctx context.Context, name, address string)
+	// TODO some filters have to be dynamic, so this has to override chain reader bind that comes from config?
+	// Maube this should just be Bind?
+	// RegisterFilter()
+	// UnRegisterFilter()
+
+	QueryKey(ctx context.Context, keys string, queryFilters []QueryFilter, limitAndSort LimitAndSort, sequenceDataType any) ([]Sequence, error)
+	QueryKeys(ctx context.Context, keys []string, queryFilters []QueryFilter, limitAndSort LimitAndSort, sequenceDataTypes []any) ([][]Sequence, error)
```

Comment:
I am initially confused on the return type here being a multi-dimensional slice. Of the existing log poller functions, none of them have this return type. They return either a single log value or a slice of values.

Reply:
Since the caller can't differentiate between sequences they have to do it by relying on dimensions, because they can know to which dimension of key and sequenceDataTypes it belongs to